### DISCUSSION TITLE
Help solve caching problem after using direct query to database

### DISCUSSION THREAD

**Discussion by mrZedov** [mikro-orm/mikro-orm#6012] [Q&A]
I execute the following code below
```
const em = this.orm.em.fork();
await em.getConnection().execute(`
   DELETE FROM public.record r
   USING public.gc_event e
   left join gc_calendar c on c.id=e.gc_calendar_id
   WHERE r.id = e.record_id and c.gc_account_id=?
`,[gcId]);
```

After that, the gc_event table is cleared of unnecessary records.
But the problem is that when adding a new record, it seems to me that the ORM does not see the changes. It tries to make an update instead of an insert. I can add a new record with the same ID that could have been before.
Perhaps using the @CreateRequestContext() decorator would solve the problem, but it seems to me that it will be wrong if I insert it everywhere.
My opinion is that something needs to be done with the cache after a direct request to delete it.
I would appreciate a hint!

**Comment by B4nan** [comment]
Few notes:
- raw queries have no effect on the state, if you have some entity in the context and you delete it via raw query, its up to you to clear the context manually (or use a fresh fork)
- you don't have to use a fork to run raw queries
- you can use `em.execute` instead of `em.getConnection().execute`

**Comment by mrZedov** [comment]
Thanks for the comments. Is it possible to clear the context after an unhandled delete request?
I use "create" in another module and would like to use the context in normal cases. I need to clear the context only after an unhandled request. I found a use for "clearCache", but it requires applying to each ID.

When I add records, but ORM sees outdated data from the cache and replaces them with "update"

```
await this.gcEventCrudService.create({
id: event.id,
gcCalendar: calendarId,
record: recordRec.id,
fromGC: true,
recurrentEventId: event.recurringEventId,
event: event,
});
```

**Reply by B4nan** [reply]
`clearCache` is for result cache, that's not the same as the identity map. you can clear the whole context via `em.clear()`, or remove a single entity via `em.getUnitOfWork().unsetIdentity(ent)`.
